locator/tablets.hh

```diff
@@ -218,6 +222,7 @@ enum class tablet_transition_stage {
     allow_write_both_read_old,
     write_both_read_old,
     streaming,
+    rebuild_repair,
```

Comment:
I think it is safer to append to the end to avoid changing the meaning of existing values.

Reply:
It's not ABI, so we keep the list aligned with the flow of stages.

Reply:
Before the value 03 is for  write_both_read_new, after the change the value 03 is for rebuild_repair.  In a mixed cluster, the old and new node will have a different interpretation for the value 03. Isn't this a problem? Why do we introduce this confusion if we could avoid it by just appending to the end of the list. 

It is nice to make the list follow the stage flows, but it is not a must for the flows to work. We can assign only numbers to the enum members.

Reply:
> Before the value 03 is for write_both_read_new, after the change the value 03 is for rebuild_repair. In a mixed cluster, the old and new node will have a different interpretation for the value 03. Isn't this a problem?

It's not a problem. Stages are stored and printed by name, so integers don't have to be stable.